### PR TITLE
Horizon scanning template

### PR DESCRIPTION
**Pull request by blanchco**
# Description

* A first pass of the horizon scanning template
* A user can select _x_ number of interventions and _y_ number of parameters with a low and high value.
* For each parameter low and high value, a new model config with be created with that value as a constant.  For example selecting beta with an extreme low of 0.1 and extreme high of 1.1, 2 configurations will be created with a constant beta of 0.1 and another with a constant beta of 1.1.
* These new configurations will each be attached to a simulate node with each intervention node.

### CODE REVIEW

File: packages/client/hmi-client/src/components/workflow/ops/compare-datasets/compare-datasets-utils.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { WorkflowNode } from '@/types/workflow';
+import { computed, Ref } from 'vue';
+import { DataArray } from '@/services/models/simulation-service';
+import _ from 'lodash';
+import { CompareDatasetsState } from './compare-datasets-operation';
+
+export function usePreparedChartInputs(
+	props: {
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/workflow/ops/compare-datasets/compare-datasets-utils.ts:8]: `usePreparedChartInputs` seems to be unrelated to the scenario template change here?

File: packages/client/hmi-client/src/components/workflow/scenario-templates/horizon-scanning/tera-horizon-scanning-template.vue [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+	</tera-scenario-template>
+</template>
+
+<script setup lang="ts">
+import { HorizonScanningScenario } from '@/components/workflow/scenario-templates/horizon-scanning/horizon-scanning-scenario';
+import { useProjects } from '@/composables/project';
+import { getInterventionPoliciesForModel, getModel, getModelConfigurationsForModel } from '@/services/model';
+import { AssetType, InterventionPolicy, ModelConfiguration, ParameterSemantic } from '@/types/Types';
+import _, { isEmpty } from 'lodash';
+import { computed, ref, watch } from 'vue';
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/workflow/scenario-templates/horizon-scanning/tera-horizon-scanning-template.vue:125]: nitpick: preferable to have the imports in the following order:
- lodash and other vendor utilities
- vue stuff
- our internal services/components ... etc

This ordering makes things a little safer in case any library decides to override stuff, eg us override lodash or whatnot.

File: packages/client/hmi-client/src/components/workflow/scenario-templates/horizon-scanning/horizon-scanning-scenario.ts [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+import { getInterventionPolicyById } from '@/services/intervention-policy';
+import { useProjects } from '@/composables/project';
+
+export interface HorizonScanningParameter {
+	id: string;
+	low: number;
+	high: number;
+}
+export class HorizonScanningScenario extends BaseScenario {
+	public static templateId = 'horizon-scanning';
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/workflow/scenario-templates/horizon-scanning/horizon-scanning-scenario.ts:25]: A short description of what is a horizon-scanning-scenario is here would be appreciated!

File: packages/client/hmi-client/src/components/workflow/ops/compare-datasets/compare-datasets-utils.ts (hunk 1 above)
Comment (blanchco) [reply packages/client/hmi-client/src/components/workflow/ops/compare-datasets/compare-datasets-utils.ts:8]: Oops, that snuck in there will remove this file

File: packages/client/hmi-client/src/components/workflow/scenario-templates/horizon-scanning/tera-horizon-scanning-template.vue (hunk 2 above)
Comment (blanchco) [reply packages/client/hmi-client/src/components/workflow/scenario-templates/horizon-scanning/tera-horizon-scanning-template.vue:125]: good to know, I've reorganized the imports here for the horizon scanning

File: packages/client/hmi-client/src/components/workflow/scenario-templates/horizon-scanning/horizon-scanning-scenario.ts (hunk 3 above)
Comment (blanchco) [reply packages/client/hmi-client/src/components/workflow/scenario-templates/horizon-scanning/horizon-scanning-scenario.ts:58]: added a blurb here

**Review by mwdchang — APPROVED** [review]
Didn't test, seemed okay.

Please note we will have different types of distributions other than just constants and uniforms coming down the pipeline!!

**Review by YohannParis — APPROVED** [review]
